python-avd/pyavd/_eos_designs/structured_config/network_services/port_channel_interfaces.py

```diff
@@ -20,122 +23,93 @@ class PortChannelInterfacesMixin(Protocol):
+        for point_to_point_service in tenant.point_to_point_services._natural_sorted():
+            for endpoint in point_to_point_service.endpoints:
+                if self.shared_utils.hostname not in endpoint.nodes:
+                    continue
+
+                node_index = endpoint.nodes.index(self.shared_utils.hostname)
+                interface_name = endpoint.interfaces[node_index]
+                if (port_channel_mode := endpoint.port_channel.mode) not in ["active", "on"]:
+                    continue
+
+                channel_group_id = "".join(re.findall(r"\d", interface_name))
+                interface_name = f"Port-Channel{channel_group_id}"
+                if point_to_point_service.subinterfaces:
+                    # This is a subinterface so we need to ensure that the parent is created
+                    parent_interface = EosCliConfigGen.PortChannelInterfacesItem(
+                        name=interface_name,
+                        peer_type="system",
+                        shutdown=False,
+                    )
+                    parent_interface.switchport.enabled = False
+
+                    if (short_esi := endpoint.port_channel.short_esi) is not None and len(short_esi.split(":")) == 3:
+                        parent_interface.evpn_ethernet_segment._update(
+                            identifier=f"{self.inputs.evpn_short_esi_prefix}{short_esi}", route_target=short_esi_to_route_target(short_esi)
                         )
+                        if port_channel_mode == "active":
+                            parent_interface.lacp_id = short_esi.replace(":", ".")
+
+                    # TODO: See to refactor
+                    subif_parent_interfaces.append(parent_interface)
 
-            port_channel_interfaces.extend(
-                subif_parent_interface for subif_parent_interface in subif_parent_interfaces if subif_parent_interface not in port_channel_interfaces
-            )
+                    for subif in point_to_point_service.subinterfaces:
+                        subif_name = f"{interface_name}.{subif.number}"
 
-        if port_channel_interfaces:
-            return port_channel_interfaces
+                        po_interface = EosCliConfigGen.PortChannelInterfacesItem(
+                            name=subif_name,
+                            peer_type="point_to_point_service",
+                            shutdown=False,
+                        )
+                        po_interface.encapsulation_vlan.client._update(encapsulation="dot1q", vlan=subif.number)
+                        po_interface.encapsulation_vlan.network.encapsulation = "client"
+
+                        self.structured_config.port_channel_interfaces.append(po_interface)
+
+                else:
+                    po_interface = EosCliConfigGen.PortChannelInterfacesItem(
+                        name=interface_name,
```

Comment:
add the names to a set to verify if autogenerated parents should be added (rather than checking in structured_config)